src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -1218,3 +1221,20 @@
     (when (fs/exists? cert)
       (fs/delete cert)
       (log/debug "Deleted certificate for" subject))))
+
+(schema/defn ^:always-validate get-custom-oid-mappings :- (schema/maybe OIDMappings)
+  "Given a path to a custom OID mappings file, return a map of all oids to
+  shortnames"
+  [custom-oid-mapping-file :- (schema/maybe schema/Str)]
+  (when (fs/file? custom-oid-mapping-file)
+    (try
+      (let [oid-mappings (:oid_mapping (yaml/parse-string (slurp custom-oid-mapping-file)))]
+        (into {} (for [[oid names] oid-mappings] [(name oid) (keyword (:shortname names))])))
+      (catch Exception e
```

Comment:
I have no idea. I took a look at `clj-yaml` but there's no documentation and no indication of whether or not the `parse-string` function will throw any exceptions.

I dug into SnakeYAML, which is what `clj-yaml` uses for all its parsing, and it doesn't look like loading a YAML file should throw any exceptions, although in that case I'm unsure how it handles invalid syntax.

It's possible the `slurp` call could end up throwing an exception?